[mjyan02] iP

mjyan02 opened the pull request:
# Duke Chatbot

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Duke frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is:

1. download it from [here](https://github.com/mjyan02/ip.git).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

**Features:**

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

---

> [!NOTE] If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```
public class Main {
    public static void main(String[] args) {
        Application.launch(DukeLauncher.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

mjyan02 committed:
Level 1

mjyan02 committed:
Level 1

mjyan02 committed:
Updated

mjyan02 committed:
Updated Bye Statement

mjyan02 committed:
Updated Bye Statement

mjyan02 committed:
Level 2

mjyan02 committed:
Level 3

mjyan02 committed:
Level 4

mjyan02 committed:
A-TextUiTesting

mjyan02 committed:
Level-5

mjyan02 committed:
Level-7

mjyan02 committed:
Level-7

mjyan02 committed:
Level-7

mjyan02 committed:
Level-6

mjyan02 committed:
Merge remote-tracking branch 'origin/branch-Level-7' into branch-Level-7

mjyan02 committed:
Level-7 (Merge branch 'master' into branch-Level-7)

mjyan02 committed:
Merge Commit

Branch Level 7 -> Master

mjyan02 committed:
Merge Commit

Branch Level 7 -> Master

mjyan02 committed:
Level-8

mjyan02 committed:
Merge pull request #2 from mjyan02/branch-Level-8

Branch level 8 -> Master

mjyan02 committed:
A-MoreOOP

mjyan02 committed:
Merge pull request #3 from mjyan02/A-MoreOOP

A-MoreOOP -> Master

mjyan02 committed:
A-Packages

mjyan02 committed:
Merge pull request #4 from mjyan02/add-gradle-support

Add gradle support -> Master

mjyan02 committed:
A-Gradle

mjyan02 committed:
A-JUnit

# Created tests for Duke.java
# Created tests for Todo.java
# Created tests for Deadline.java
# Created tests for Event.java

mjyan02 committed:
A-Jar

mjyan02 committed:
Merge remote-tracking branch 'origin/master'

mjyan02 committed:
A-JavaDoc

mjyan02 committed:
A-CodingStandard

mjyan02 committed:
Level-9

mjyan02 committed:
Merge pull request #5 from mjyan02/branch-A-JavaDoc

A-JavaDoc -> Master

mjyan02 committed:
Merge branch 'master' into branch-A-CodingStandard

mjyan02 committed:
Merge pull request #6 from mjyan02/branch-A-CodingStandard

A-CodingStandard -> Master

mjyan02 committed:
Merge branch 'master' into branch-Level-9

mjyan02 committed:
Merge pull request #7 from mjyan02/branch-Level-9

Level-9 -> Master

mjyan02 committed:
Level-9

# Added tests for findTask() in Duke.java

mjyan02 committed:
Update Ui

mjyan02 committed:
Duke.Ui.Ui: update welcomeMessage()

mjyan02 committed:
Duke.Ui.Ui: add new line to welcomeMessage()

mjyan02 committed:
Duke.Duke: separate catch arguments to display DukeException

mjyan02 committed:
TaskList.java: update findTasks() displayMessage for no tasks found

mjyan02 committed:
TaskList.java: update listTasks() to return a String representation of taskList

DukeTest.java: update tests to use listTask() in assertEqual()

mjyan02 committed:
TaskList.java: update listTasks() to return a String representation of taskList

TaskList.java: remove getTasks()

DukeTest.java: update tests to use listTask() in assertEqual()

mjyan02 committed:
Duke.java: update switch statement indentation and fallthrough comments

mjyan02 committed:
Update Duke.java file

chinpcbenjamin commented on src/main/java/Duke/Task/Deadline.java:
```diff
@@ -0,0 +1,72 @@
+     */
+    @Override
+    public String toFile() {
+        return "D | " + (isDone ? "1" : "0") + " | " + description + " | " + by;
```
Is it better to separate this into multiple lines? It could improve code readability by using having each "+" term on a new line.

Alternatively, String.format could be useful too! 😄 

```suggestion
        return "D | "
            + (isDone ? "1" : "0")
            + " | "
            + description
            + " | "
            + by;
```

or even:

```suggestion
        return String.format("D | %d | %s | %s", isDone ? 1 : 0, description, by);
```

chinpcbenjamin commented on src/main/java/Duke/Task/Deadline.java:
```diff
@@ -0,0 +1,72 @@
+    /**
+     * The due date of the task.
+     */
+    protected LocalDate by;
```
Would it be better to use another name for this? The word "by" does not tell me a lot about what the variable is for. Perhaps it could be changed to be another name such as "dueDate" 👍

chinpcbenjamin commented on src/main/java/Duke/Task/Event.java:
```diff
@@ -0,0 +1,82 @@
+    /**
+     * The start date of the event.
+     */
+    protected LocalDate from;
+
+    /**
+     * The end date of the event.
+     */
+    protected LocalDate to;
```
Would it be better to use another name for these? The words "by" and "from" do not tell me a lot about what the variables iare for. Perhaps it could be changed to be another name such as "dueDate" and "beginDate". 👍

chinpcbenjamin reviewed (commented):
Looks good 👍 . The code was easy to read, perhaps except for the various toString methods.

mjyan02 committed:
Add: DukeGui file

mjyan02 committed:
Modify: Ui

mjyan02 committed:
DialogBox: update Ui

mjyan02 committed:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	src/main/resources/styles/styles.css

mjyan02 committed:
DialogBox: update Ui

mjyan02 committed:
DukeGui: update ui settings

mjyan02 committed:
DukeGui: update ui settings

mjyan02 committed:
Merge remote-tracking branch 'origin/branch-Level-10' into branch-Level-10

mjyan02 committed:
Merge pull request #8 from mjyan02/branch-Level-10

Branch level 10 -> master

mjyan02 committed:
Merge pull request #8 from mjyan02/branch-Level-10

Branch level 10 -> master

reaneechua commented on src/main/java/Duke/Exceptions/DukeException.java:
```diff
@@ -0,0 +1,15 @@
+package duke.exceptions;
```
Is this class necessary? Exception on its own may be sufficient already.

reaneechua commented on src/main/java/Duke/Task/Keywords.java:
```diff
@@ -0,0 +1,70 @@
+package duke.task;
```
I liked the way this class was created just to keep the keywords as enums, helps to mitigate user errors.

reaneechua commented on src/main/java/Duke/Duke.java:
```diff
@@ -0,0 +1,96 @@
+
+            Keywords keyword = Parser.parseKeyword(input);
+
+            switch (keyword) {
```
Maybe this logic could be passed to the Parser class instead? Seems like there is abit of passing back and forth between the main class and Parser, it might be possible for Parser to handle this logic.

reaneechua reviewed (commented):
Overall, I think your code was well-written and logic was sound. I didn't spot any coding violations. I feel that the class structures could be improved, but you did great! Keep up the good work!

mjyan02 committed:
Merge remote-tracking branch 'origin/master'

mjyan02 committed:
Ui classes: display a Gui for Duke Class

Duke is launched using DukeLauncher to create a GUI for Duke.

The common behaviors across different types of Task cause code duplication.

Extracting the common behavior into a super class allows us to use
polymorphism when dealing with task attributes.

Let's pull up behaviors common to all task attribute classes into a new
parent class named Task.

Using inheritance is preferable over composition in this situation
because the common behaviors are not composable.

mjyan02 committed:
Merge remote-tracking branch 'origin/master'

mjyan02 committed:
Duke: update Duke.java to include Assertions when required

mjyan02 committed:
Merge remote-tracking branch 'origin/branch-A-Assertions' into branch-A-Assertions

mjyan02 committed:
TaskList: modify TaskList.java to follow proper coding quality

mjyan02 committed:
TaskList: modify TaskList.java to follow proper coding quality

mjyan02 committed:
Merge pull request #10 from mjyan02/branch-A-Assertions

Branch-a-assertions

mjyan02 committed:
Merge remote-tracking branch 'origin/master' into branch-A-CodeQuality

mjyan02 committed:
TaskList: modify TaskList.java to follow proper coding quality

mjyan02 committed:
TaskList: modify TaskList.java to follow proper coding quality

mjyan02 committed:
Merge: merge master into branch-A-CodeQuality

mjyan02 committed:
Merge: merge master into branch-A-CodeQuality

mjyan02 committed:
Merge remote-tracking branch 'origin/branch-A-CodeQuality' into branch-A-CodeQuality

mjyan02 committed:
Merge pull request #11 from mjyan02/branch-A-CodeQuality

Branch a code quality

mjyan02 committed:
Merge branch 'branch-A-CodeQuality' of https://github.com/mjyan02/ip into branch-A-CodeQuality

mjyan02 committed:
Update: update TaskList.java to include feature to detect duplicate task entries

mjyan02 committed:
Modify: update parser class to include extractDescription method and fix bugs in TaskList for task description

mjyan02 committed:
Merge remote-tracking branch 'origin/master'

mjyan02 committed:
Modify: update parser class to include extractDescription method and fix bugs in TaskList for task description

mjyan02 committed:
Modify: update README.md in docs folder

mjyan02 committed:
Modify: change name of Chatbot from Duke to Zirnitra and added Ui.png into docs

mjyan02 committed:
Modify: update README.md in docs folder

mjyan02 committed:
Merge remote-tracking branch 'origin/master'

mjyan02 committed:
Modify: update README.md in docs folder

mjyan02 committed:
Modify: update README.md in docs folder

mjyan02 committed:
Modify: fix Storage filepath to create new TaskList file in working directory

mjyan02 committed:
Modify: fix Storage filepath to create new TaskList file in working directory

mjyan02 committed:
Modify: update README.md in docs folder

mjyan02 committed:
Modify: update README.md in docs folder

mjyan02 committed:
Merge remote-tracking branch 'origin/master'

mjyan02 committed:
Modify: update README.md in docs folder

mjyan02 committed:
Modify: update README.md in docs folder

mjyan02 committed:
Modify: update ZirnitraGui to show Message Zirnitra in dialogbox

mjyan02 committed:
Modify: update Ui.png screenshot

mjyan02 committed:
Remove: delete extra files

mjyan02 committed:
Remove: delete extra files

mjyan02 committed:
Merge remote-tracking branch 'origin/master'

mjyan02 committed:
Modify: update ui package to use .fxml files instead

mjyan02 committed:
Modify: remove extra test files

mjyan02 committed:
Modify: remove emoji from welcomeMessage()

mjyan02 committed:
Modify: fix bugs from Main.java and MainWindow.java for jar file to work properly

mjyan02 committed:
Modify: update Javadocs in src files

mjyan02 committed:
Modify: update Javadocs in src files

mjyan02 committed:
Modify: update Javadocs comments

mjyan02 committed:
Modify: update DialogBox error handling

mjyan02 committed:
Modify: update DialogBox error handling

mjyan02 committed:
Modify: update DialogBox Javadocs

mjyan02 committed:
Modify: update DialogBox Javadocs

mjyan02 committed:
Modify: update Javadocs across files
